server/src/main/java/io/crate/types/ObjectType.java

```diff
@@ -78,17 +81,40 @@ public ValueIndexer<Map<String, Object>> valueIndexer(RelationName table,
         }
     };
 
+    public static ObjectType of(ColumnPolicy columnPolicy) {
```

Comment:
I'm not sure if we want this.  Shouldn't most objectType creation code go through the builder to also set inner types? 

We probably shouldn't make it too convenient to use empty instances. Same goes for the constants. Do we really need all 3?